src/main/kotlin/org/zowe/kotlinsdk/zowe/config/ZoweConfig.kt

```diff
@@ -188,12 +191,25 @@ class ZoweConfig(
    */
   fun PropertyBuilder.set (value: Any?) {
     val profiles = profilesToSearchProp.filterNotNull()
+    var isSet = false
     for (profile in profiles) {
-      if (profile.properties?.containsKey(propName) == true) {
-        profile.properties[propName] = value
+      //try to set main profile
+      if (profile.type != "base" && !isSet) {
+        if (profile.properties?.containsKey(propName) == true) {
+          profile.properties[propName] = value
+          isSet = true
+        }
```

Comment:
no need to nest + use `.forEach` instead of `for` loop with early return when the `profile.properties[propName]` is set